app/src/main/java/com/example/unlibrary/util/AuthenticatedActivity.java

```diff
@@ -36,21 +36,40 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        FirebaseUser mUser = FirebaseAuth.getInstance().getCurrentUser();
+        mUser = FirebaseAuth.getInstance().getCurrentUser();
```

Comment:
```suggestion
        FirebaseAuth.getInstance().addAuthStateListener(firebaseAuth -> {
            if (firebaseAuth.getCurrentUser() == null) {
                goToAuth();
                return;
            }

            mUser = FirebaseAuth.getInstance().getCurrentUser();
        });
```
Alternative, this way we can just call `FirebaseAuth.getInstance().signOut();` and not be concerned about propagating back to MainActivity